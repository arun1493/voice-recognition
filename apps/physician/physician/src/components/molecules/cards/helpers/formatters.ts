import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
