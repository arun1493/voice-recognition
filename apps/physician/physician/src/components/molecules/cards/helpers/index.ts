export * from "./labels";
export * from "./formatters";
