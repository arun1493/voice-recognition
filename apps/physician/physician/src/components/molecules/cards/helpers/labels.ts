import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
