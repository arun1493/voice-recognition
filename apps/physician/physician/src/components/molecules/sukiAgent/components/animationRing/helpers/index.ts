export * from "./ringSize";
export * from "./animation";
