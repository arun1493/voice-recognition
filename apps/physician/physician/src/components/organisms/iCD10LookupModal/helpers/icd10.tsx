import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
