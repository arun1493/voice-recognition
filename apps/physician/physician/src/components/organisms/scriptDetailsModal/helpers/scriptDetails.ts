import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
