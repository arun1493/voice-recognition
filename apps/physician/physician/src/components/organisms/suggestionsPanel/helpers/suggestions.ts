import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
