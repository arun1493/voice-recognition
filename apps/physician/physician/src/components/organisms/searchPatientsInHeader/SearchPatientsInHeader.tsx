import _get from "lodash/get";
import _noop from "lodash/noop";
import _isEmpty from "lodash/isEmpty";
import _debounce from "lodash/debounce";
import { useNavigate } from "react-router-dom";
import { useQuery } from "@tanstack/react-query";
import { FunctionComponent, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Modal, useBreakpoint } from "@suki-web/suki-ui";

import { Patient } from "../../../types";
import { patientReader } from "../../../readers";
import { selectEmrState } from "../../../redux/emr";
import { logger, searchPatient } from "../../../api";
import { modalsActions } from "../../../redux/modals";
import { StyledModalContent } from "./searchPatientsInHeader.styled";
import { GlobalPatientsSearch } from "../../molecules/globalPatientsSearch";
import { getRoute, shouldShowEmrIntegrationToolbar } from "../../../helpers";
import { getSearchPatientsModalMargins } from "./searchPatientsInHeader.helpers";
import { Pages, QUERY_NAMES, SEARCH_PATIENTS_RESULTS_LIMIT, searchPatientsModalWidth } from "../../../constants";

const SearchPatientsInHeader: FunctionComponent = () => {
  const breakpoint = useBreakpoint();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const [searchValue, setSearchValue] = useState("");

  const emrState = useSelector(selectEmrState);
  const isEmrPanelShown = shouldShowEmrIntegrationToolbar(emrState);

  const { data, isFetching } = useQuery(
    [QUERY_NAMES.SEARCH_PATIENTS, searchValue, SEARCH_PATIENTS_RESULTS_LIMIT],
    () => searchPatient(searchValue, SEARCH_PATIENTS_RESULTS_LIMIT),
    {
      enabled: !_isEmpty(searchValue),
      onError: (error) => logger.error({ message: "error fetching patients", payload: error })
    }
  );
  const patients = _get(data, "data.results");

  const handleClose = () => {
    dispatch(modalsActions.hideSearchPatientsInHeaderModal());
  };

  const handleSearchInput = _debounce(async (value: string) => {
    setSearchValue(value);
  }, 200);

  const handleGoToPatientPage = (selectedPatient: Patient) => {
    const patientId = patientReader.id(selectedPatient);
    if (patientId) {
      const patientProfileRoute = getRoute(Pages.PatientProfile, { patientId });
      patientProfileRoute && navigate(patientProfileRoute);
      handleClose();
    }
  };

  return (
    <Modal
      closeDuration={1}
      hasMask={false}
      headerHeight={40}
      height="auto"
      isMaskClosable={true}
      isMaskTransparent={true}
      isWithTooltip={false}
      margin={getSearchPatientsModalMargins(breakpoint, isEmrPanelShown)}
      onClose={_noop}
      onModalClose={handleClose}
      position="tc"
      testId="search-patients-in-header"
      width={searchPatientsModalWidth[breakpoint]}
    >
      <StyledModalContent hasPadding={false}>
        <GlobalPatientsSearch
          hasBorder={false}
          hasCloseInInput={true}
          hasShadow={false}
          onPatientClick={handleGoToPatientPage}
          onSearch={handleSearchInput}
          patients={patients}
          prefixIcon={isFetching ? "loader" : "search"}
          value={searchValue}
        />
      </StyledModalContent>
    </Modal>
  );
};

export { SearchPatientsInHeader };

export default SearchPatientsInHeader;
