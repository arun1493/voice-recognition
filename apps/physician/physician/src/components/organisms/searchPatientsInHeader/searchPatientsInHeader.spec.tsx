import _noop from "lodash/noop";
import ReactDOM from "react-dom";
import { ReactPortal } from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";

import { configureI18n } from "../../../helpers";
import { componentSetupWithStore } from "../../../hocs";
import { SearchPatientsInHeader } from "./SearchPatientsInHeader";
import { shouldMatchSnapshot } from "../../../helpers/snapshot.spec";

const setup = componentSetupWithStore(() => <SearchPatientsInHeader />);

describe("Search Patients in header", () => {
  let baseElement: Element;

  beforeEach(() => {
    configureI18n();
    baseElement = setup().baseElement;
    ReactDOM.createPortal = jest.fn<ReactPortal, any>((element, node) => element);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    expect(baseElement).toBeTruthy();
  });

  it("should have one input field", () => {
    const input = baseElement.getElementsByTagName("input");
    expect(input.length).toEqual(1);
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(SearchPatientsInHeader);
  });
});
