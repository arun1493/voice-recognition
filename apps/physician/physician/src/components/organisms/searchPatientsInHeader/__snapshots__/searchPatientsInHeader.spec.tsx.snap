// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Patients in header should match snapshot 1`] = `
<div
  className="modalstyled__StyledModal-sc-1u144rj-1 hwiWDf"
  data-cy="search-patients-in-header"
  height="auto"
  width="508px"
>
  <div
    aria-label="Close"
    className="modalstyled__StyledCloseButton-sc-1u144rj-0 cotaWZ modal-close"
    height={40}
  >
    <div
      className="navigationIconstyled__StyledNavigationIcon-lwvkbw-0 imuqWK"
      data-cy=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        className="iconstyled__StyledButton-ifk1iv-0 kgKJBV"
        onClick={[Function]}
      >
        <svg
          height={24}
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.70857 6.29289C7.31805 5.90237 6.68488 5.90237 6.29436 6.29289C5.90383 6.68342 5.90383 7.31658 6.29436 7.70711L10.5865 11.9993L6.2944 16.2914C5.90388 16.6819 5.90388 17.3151 6.2944 17.7056C6.68493 18.0961 7.31809 18.0961 7.70862 17.7056L12.0007 13.4135L16.2928 17.7056C16.6834 18.0961 17.3165 18.0961 17.7071 17.7056C18.0976 17.3151 18.0976 16.6819 17.7071 16.2914L13.4149 11.9993L17.7071 7.70711C18.0976 7.31658 18.0976 6.68342 17.7071 6.29289C17.3166 5.90237 16.6834 5.90237 16.2929 6.29289L12.0007 10.5851L7.70857 6.29289Z"
            fill="#999999"
            xmlns="http://www.w3.org/2000/svg"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="modalContentstyled__StyledModalContent-wawizf-0 fzECCo searchPatientsInHeaderstyled__StyledModalContent-sc-1lbsrag-0 ecqmrp"
    height="auto"
    onScroll={[Function]}
  >
    <div
      className="globalPatientsSearchstyled__StyledPatientsSearch-nftnqf-0 eIhoVw searchPatientsInHeaderstyled__StyledGlobalPatientsSearch-sc-1lbsrag-1 fSXCBJ"
    >
      <div
        className="globalPatientsSearchstyled__StyledSearchContainer-nftnqf-1 BuUsd"
      >
        <div
          className="globalPatientsSearchstyled__StyledSearchIcon-nftnqf-3 tfHfz"
        >
          <i
            className="SukiIcon__StyledIcon-q9b9ar-0 hViFDz sukiIcon "
            color="secondary"
            size="md"
          >
            <svg
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.842 3.043a6.191 6.191 0 0 0-2.022.548c-1.972.898-3.326 2.651-3.73 4.829-.092.495-.092 1.665 0 2.16.514 2.771 2.559 4.816 5.33 5.33.495.092 1.665.092 2.16 0 2.771-.514 4.816-2.559 5.33-5.33.092-.495.092-1.665 0-2.16-.451-2.433-2.09-4.326-4.41-5.095a6.485 6.485 0 0 0-2.658-.282m1.218 2.001a4.52 4.52 0 0 1 3.48 2.493c.721 1.489.559 3.304-.411 4.606-.582.782-1.313 1.319-2.2 1.616a4.4 4.4 0 0 1-2.858 0c-1.981-.664-3.253-2.632-3.038-4.699a4.484 4.484 0 0 1 5.027-4.016m3.86 10.471c-.3.098-.554.326-.671.602-.076.181-.075.542.002.729.065.154 2.931 4.263 3.099 4.442.252.269.742.359 1.096.202.217-.096.423-.298.519-.509.09-.197.097-.57.016-.765-.069-.165-2.893-4.212-3.077-4.41a1.26 1.26 0 0 0-.281-.214c-.172-.088-.544-.129-.703-.077"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </i>
        </div>
        <div
          className="formInputstyled__StyledInputContainer-p3pq9x-0 jOohTQ"
        >
          <input
            className="formInputstyled__StyledFormInput-p3pq9x-1 ckqHZa globalPatientsSearchstyled__StyledSearchInput-nftnqf-2 daWdeC"
            data-cy="form-input"
            data-testid="input"
            disabled={false}
            onChange={[Function]}
            placeholder="Enter Patient Name"
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
